lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_online_resource.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'adiwg/mdtranslator/internal/internal_metadata_obj'
+
+module ADIWG
+   module Mdtranslator
+      module Readers
+         module Iso191152
+            module OnlineResource
+               @@ciOnlineResourceXPath = 'gmd:CI_OnlineResource'
+               @@linkageXPath = 'gmd:linkage//gmd:URL'
+               @@protocolXPath = 'gmd:protocol//gco:CharacterString'
+               @@appProfileXPath = 'gmd:applicationProfile//gco:CharacterString'
+               @@nameXPath = 'gmd:name//gco:CharacterString'
+               @@descXPath = 'gmd:description//gco:CharacterString'
+               @@funcXPath = 'gmd:function//gmd:CI_OnLineFunctionCode'
+               @@protocolReqXPath = 'gmd:protocolRequest//gco:CharacterString'
+               def self.unpack(xParent, _hResponseObj)
+                  # xParent because an online resource can occur within
+                  # mcc:linkage or gmd:onlineResource
+                  intMetadataClass = InternalMetadata.new
+                  hOnlineResource = intMetadataClass.newOnlineResource
+
+                  xCiOnelineResource = xParent.xpath(@@ciOnlineResourceXPath)
+
+                  return nil if xCiOnelineResource.empty?
+
+                  # :olResURI (optional)
```

Comment:
this is actually required in -2 & -3. do you mind updating -3 as well in this PR? that's an oversight on my part. also, can you add the xsd element as evidence?

Reply:
you don't have to include a error message test for -3 since that group of tests doesn't really exit for that reader. there's a ticket to add them though. i'll add a comment on that ticket about this